server/market/market.go

```diff
@@ -790,6 +790,21 @@ func (m *Market) Run(ctx context.Context) {
 		}
 		m.persistBook = true // future resume default
 		m.activeEpochIdx = 0
+
+		// Revoke any unmatched epoch orders (if context was canceled, not a
+		// clean suspend stopped the market).
+		for oid, ord := range m.epochOrders {
+			log.Infof("Dropping epoch order %v", oid)
+			if co, ok := ord.(*order.CancelOrder); ok {
+				if err := m.storage.FailCancelOrder(co); err != nil {
+					log.Errorf("Failed set orphaned epoch cancel order %v as executed: %v", oid, err)
+				}
+				continue
+			}
+			if err := m.storage.ExecuteOrder(ord); err != nil {
+				log.Errorf("Failed set orphaned epoch trade order %v as executed: %v", oid, err)
+			}
+		}
```

Comment:
The suspend command should be used to complete the epoch, but ctrl+c should at least not ditch orders in epoch status even though they do not go through matching.

Reply:
Just noticing the difference between the executed status here, and the revoked status when we do a book purge. Seems like similar situations.

Reply:
Yeah, maybe revoked is better for the trades too.  Will switch it.  Really just avoiding eternal epoch status orders though.

Reply:
OK, so we could use RevokeOrder{Uncounted} for the trade order, but that creates a cancel order to go with it.  The general idea of order revocation is that it's a forced unbooking, a server-generated cancellation.  So while executed may not fix exactly here going from epoch->executed is more inline with how this is handled elsewhere, such as epoch orders not matching becoming executed, not revoked.

Reply:
Would an operator have any way to know by looking at the order row in the DB whether this order was executed via matching vs. dirty shutdown?

Reply:
No preimage.

Reply:
Hmm, need to double check that these won't count as a preimage miss, but had be pretty darn uncommon to do a dirty shutdown without suspending the markets first.

Reply:
OK, so by making it executed we also escape having it counted as a preimage miss because the preimage miss query looks for **revoked** status orders.
Still thinking through other pitfalls with this though.